src/org/labkey/test/tests/visualization/AssayRunFilterTest.java

```diff
@@ -0,0 +1,194 @@
+
+        // view the unfilteredChart, with a filter on run
+        AssayRunsPage runsPage = goToAssayRunsPage(assayName);
+        AssayDataPage dataPage = runsPage.clickAssayIdLink("firstRun");
+        dataPage.getDataTable().clickReportMenu(false,"HeightWeightAll");
+        String expected = "848688909294168170172174176178180182184186188Dataweightheight";
+        assertSVG(expected, 0);
+
+        // now filter on the other run and verify expected results
+        dataPage = goToAssayRunsPage(assayName).clickAssayIdLink("secondRun");
+        String secondRunExpected = "63.56464.56565.56666.567171172173174175176177178Dataweightheight";
+        dataPage.getDataTable().clickReportMenu(false,"HeightWeightAll");
+        assertSVG(secondRunExpected);
+
+        // now verify filter built into a chart will be applied when also filtering based on run
+        // create filter to show only firstRun results
+        goToAssayRunsPage(assayName)
+                .clickAssayIdLink("firstRun")
+                .getDataTable()
+                .createChart()
+                .setChartType(ChartTypeDialog.ChartType.Scatter)
+                .setYAxis("height")
+                .setXAxis("weight")
+                .clickApply()
+                .clickSave()
+                .setReportName("HeightWeightFirstRun").clickSave();
+        // now filter to see only secondRun results
+        goToAssayRunsPage(assayName)
+                .clickAssayIdLink("secondRun")
+                .getDataTable()
+                .clickReportMenu(false, "HeightWeightFirstRun");
+        // expect both filters to result in 0 records shown
+        waitForText("The response returned 0 rows of data.");
+        assertSVG("00.20.40.60.8100.10.20.30.40.50.60.70.80.91Dataweightheight");
+    }
+
+    private Protocol getProtocol(String assayName, String providerName) throws Exception
+    {
+        // get the template from the server
+        Connection cn = createDefaultConnection();
+        String containerPath = getProjectName();
+        GetProtocolCommand getProtocolCommand = new GetProtocolCommand(providerName);
+        ProtocolResponse getProtocolResponse = getProtocolCommand.execute(cn, containerPath);
+
+        // modify the protocol according to our needs
+        Protocol newAssayProtocol = getProtocolResponse.getProtocol();
+        newAssayProtocol.setName(assayName);
+
+        return newAssayProtocol;
+    }
+
+    private Protocol saveProtocol(Connection cn, Protocol protocol) throws Exception
+    {
+        SaveProtocolCommand saveProtocolCommand = new SaveProtocolCommand(protocol);
+        ProtocolResponse savedProtocolResponse = saveProtocolCommand.execute(cn, getProjectName());
+        return savedProtocolResponse.getProtocol();
+    }
+
+    private AssayRunsPage goToAssayRunsPage(String assayName)
+    {
+        goToProjectHome();
+        clickAndWait(Locator.linkWithText("Assay List"));
```

Comment:
Are there not utilities/components to do this navigation to an assay by name?

Reply:
If there are, I don't know about them.  We support this in biologics, but apparently not in LKS

Reply:
Not by name.  In LKS you navigate to the assay by knowing the protocol's Id.
There is an APIAssayHelper that promises to get a protocolId from the name, but we have the ID already in the test context.